cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -323,7 +332,7 @@ String getImageVersion(ProvisionerContext context, DataprocConf conf) {
         default:
           if (context.getAppCDAPVersionInfo() == null ||
             context.getAppCDAPVersionInfo().compareTo(SPARK3_CDAP_DEFAULT) < 0) {
```

Comment:
looks like there's already some VersionInfo interface that's used here for comparison. Maybe we can also add some method to parse a string into a VersionInfo instead of using the Hadoop one? I'm a bit hesitant to add more dependencies on hadoop-common in case we decide to remove it in the future.

Reply:
I have implemented a new DataprocVersionInfo class to do this comparison. PTAL